### PR TITLE
Keep obsolete license values to prevent crashe after upgrade

### PR DESCRIPTION
**Pull request by whym**
A follow-up to #478, trying to solve #506.

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/settings/Prefs.java [hunk 1]
```diff
@@ -12,5 +12,9 @@
         public static final String CC_BY_SA_4 = "CC BY-SA 4.0";
         public static final String CC_BY_4 = "CC BY 4.0";
         public static final String CC0 = "CC0";
+
+        // kept for backward compatibility to v2.1
+        @Deprecated public static final String CC_BY = "CC BY";
+        @Deprecated public static final String CC_BY_SA = "CC BY-SA";
```
Comment (whym) [inline app/src/main/java/fr/free/nrw/commons/settings/Prefs.java]: These are values from v.2.1: https://github.com/commons-app/apps-android-commons/blob/v2.1/app/src/main/java/fr/free/nrw/commons/Prefs.java

File: app/src/main/java/fr/free/nrw/commons/settings/SettingsFragment.java [hunk 2]
```diff
@@ -19,6 +19,22 @@ public void onCreate(Bundle savedInstanceState) {
...
+                getString(R.string.license_name_cc_by_sa_4_0)
+        });
+        licensePreference.setEntryValues(new String[]{
+                Prefs.Licenses.CC0,
+                Prefs.Licenses.CC_BY_3,
+                Prefs.Licenses.CC_BY_4,
+                Prefs.Licenses.CC_BY_SA_3,
+                Prefs.Licenses.CC_BY_SA_4
+        });
+
```
Comment (whym) [inline app/src/main/java/fr/free/nrw/commons/settings/SettingsFragment.java]: Revived from https://github.com/commons-app/apps-android-commons/blob/v2.1/app/src/main/java/fr/free/nrw/commons/SettingsActivity.java#L36

**Comment by maskaravivek** [comment]
LGTM (Y)

**Comment by whym** [comment]
It looks like some cases of crash still happen around Settings. I'll see if I can quickly fix them.

**Comment by domdomegg** [comment]
Looks good to me too

**Pull request merged**